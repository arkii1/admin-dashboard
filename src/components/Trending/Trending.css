.trending {
  -ms-grid-column: 2;
  grid-column: 2;
  -ms-grid-row: 2;
  grid-row: 2;
  background-color: transparent;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  gap: 1rem;
  height: 100%;
}

.trending__header {
  font-size: 2rem;
  font-weight: bold;
}

.trending__items-container {
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  gap: 1rem;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  background-color: #ffffff;
  border-radius: 10px;
  padding: 2rem;
  -webkit-box-shadow: 5px 5px 4px 2px rgba(0, 0, 0, 0.2);
          box-shadow: 5px 5px 4px 2px rgba(0, 0, 0, 0.2);
}

.trending__item {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  gap: 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.trending__item:hover {
  cursor: pointer;
  -webkit-transform: scale(1.1);
          transform: scale(1.1);
}

.trending__item__text-div {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  gap: 0.5rem;
}

.trending__item__header {
  font-size: 1.5rem;
}

.trending__item__p {
  font-size: 1.25rem;
  color: #636363;
}

.trending__avatar-credit {
  text-align: center;
  font-size: 1rem;
  font-style: italic;
  font-weight: bold;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.trending__avatar-credit:hover {
  cursor: pointer;
  -webkit-transform: scale(1.1);
          transform: scale(1.1);
}
/*# sourceMappingURL=Trending.css.map */