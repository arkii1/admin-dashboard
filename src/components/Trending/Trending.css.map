{
    "version": 3,
    "mappings": "AAEA,AAAA,SAAS,CAAC;EACN,WAAW,EAAE,CAAC;EACd,QAAQ,EAAE,CAAC;EACX,gBAAgB,EAAE,WAAW;EAC7B,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,GAAG,EAAE,IAAI;EACT,MAAM,EAAE,IAAI;CACf;;AAED,AAAA,iBAAiB,CAAC;EACd,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;CACpB;;AAED,AAAA,0BAA0B,CAAC;EACvB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,GAAG,EAAE,IAAI;EACT,eAAe,EAAE,UAAU;EAC3B,gBAAgB,ECvBV,OAAO;EDwBb,aAAa,EAAE,IAAI;EACnB,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,GAAG,CAAC,GAAG,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAe;CAC9C;;AAED,AAAA,eAAe,CAAC;EACZ,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;EACvB,GAAG,EAAE,IAAI;EACT,UAAU,EAAE,IAAI;CACnB;;AAED,AAAA,eAAe,AAAA,MAAM,CAAC;EAClB,MAAM,EAAE,OAAO;EACf,SAAS,EAAE,UAAU;CACxB;;AAED,AAAA,yBAAyB,CAAC;EACtB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,aAAa;EAC9B,GAAG,EAAE,MAAM;CACd;;AAED,AAAA,uBAAuB,CAAC;EACpB,SAAS,EAAE,MAAM;CACpB;;AAED,AAAA,kBAAkB,CAAC;EACf,SAAS,EAAE,OAAO;EAClB,KAAK,ECpDK,OAAO;CDqDpB;;AAED,AAAA,wBAAwB,CAAC;EACrB,UAAU,EAAE,MAAM;EAClB,SAAS,EAAE,IAAI;EACf,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,IAAI;CACnB;;AAED,AAAA,wBAAwB,AAAA,MAAM,CAAC;EAC3B,MAAM,EAAE,OAAO;EACf,SAAS,EAAE,UAAU;CACxB",
    "sources": [
        "Trending.scss",
        "../../varaibles.scss"
    ],
    "names": [],
    "file": "Trending.css"
}