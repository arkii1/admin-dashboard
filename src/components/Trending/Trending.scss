@import "../../varaibles.scss";

.trending {
    grid-column: 2;
    grid-row: 2;
    background-color: transparent;
    display: flex;
    flex-direction: column;
    gap: 1rem;
    height: 100%;
}

.trending__header {
    font-size: 2rem;
    font-weight: bold;
}

.trending__items-container {
    height: 100%;
    display: flex;
    flex-direction: column;
    gap: 1rem;
    justify-content: flex-start;
    background-color: $c-white;
    border-radius: 10px;
    padding: 2rem;
    box-shadow: 5px 5px 4px 2px rgba(0,0,0,0.2);
}

.trending__item {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 1rem;
    transition: 0.1s;
}

.trending__item:hover {
    cursor: pointer;
    transform: scale(1.1);
}

.trending__item__text-div {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    gap: 0.5rem;
}

.trending__item__header {
    font-size: 1.5rem;
}

.trending__item__p {
    font-size: 1.25rem;
    color: $c-dark-grey;
}

.trending__avatar-credit {
    text-align: center;
    font-size: 1rem;
    font-style: italic;
    font-weight: bold;
    transition: 0.1s;
}

.trending__avatar-credit:hover {
    cursor: pointer;
    transform: scale(1.1)
}