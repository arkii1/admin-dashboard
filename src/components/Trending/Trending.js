import React from "react"
import "./Trending.css"
import catAvatar from "../../assets/images/cat_avatar.svg"
import koalaAvatar from "../../assets/images/koala_avatar.svg"
import rhinoAvatar from "../../assets/images/rhino_avatar.svg"
import octopusAvatar from "../../assets/images/octopus_avatar.svg"

function Trending() {
  return (
    <div className="trending">
      <h1 className="trending__header">Trending</h1>
      <div className="trending__items-container">
        <div className="trending__item">
          <img src={koalaAvatar} alt="" className="avatar--small" />
          <div className="trending__item__text-div">
            <h2 className="trending__item__header">@tegan</h2>
            <p className="trending__item__p">World Peace Builder</p>
          </div>
        </div>
        <div className="trending__item">
          <img src={catAvatar} alt="" className="avatar--small" />
          <div className="trending__item__text-div">
            <h2 className="trending__item__header">@morgan</h2>
            <p className="trending__item__p">Super Cool Project</p>
          </div>
        </div>
        <div className="trending__item">
          <img src={rhinoAvatar} alt="" className="avatar--small" />
          <div className="trending__item__text-div">
            <h2 className="trending__item__header">@kendall</h2>
            <p className="trending__item__p">Life Changing App</p>
          </div>
        </div>
        <div className="trending__item">
          <img src={octopusAvatar} alt="" className="avatar--small" />
          <div className="trending__item__text-div">
            <h2 className="trending__item__header">@alex</h2>
            <p className="trending__item__p">No Traffic Maker</p>
          </div>
        </div>
        <a
          href="https://www.vecteezy.com/free-vector/animal-avatar"
          className="trending__avatar-credit"
        >
          Animal Avatar Vectors by Vecteezy
        </a>
      </div>
    </div>
  )
}

export default Trending
