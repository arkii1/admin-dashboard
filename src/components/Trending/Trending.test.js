import React from "react"
import ReactDOM from "react-dom/client"
import { cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Trending from "./Trending"

afterEach(cleanup)

it("renders without crashing", () => {
  const container = document.createElement("div")
  const root = ReactDOM.createRoot(container)
  root.render(<Trending />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<Trending />).toJSON()
  expect(tree).toMatchSnapshot()
})
