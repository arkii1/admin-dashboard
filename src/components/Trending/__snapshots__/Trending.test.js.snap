// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="trending"
>
  <h1
    className="trending__header"
  >
    Trending
  </h1>
  <div
    className="trending__items-container"
  >
    <div
      className="trending__item"
    >
      <img
        alt=""
        className="avatar--small"
        src="koala_avatar.svg"
      />
      <div
        className="trending__item__text-div"
      >
        <h2
          className="trending__item__header"
        >
          @tegan
        </h2>
        <p
          className="trending__item__p"
        >
          World Peace Builder
        </p>
      </div>
    </div>
    <div
      className="trending__item"
    >
      <img
        alt=""
        className="avatar--small"
        src="cat_avatar.svg"
      />
      <div
        className="trending__item__text-div"
      >
        <h2
          className="trending__item__header"
        >
          @morgan
        </h2>
        <p
          className="trending__item__p"
        >
          Super Cool Project
        </p>
      </div>
    </div>
    <div
      className="trending__item"
    >
      <img
        alt=""
        className="avatar--small"
        src="rhino_avatar.svg"
      />
      <div
        className="trending__item__text-div"
      >
        <h2
          className="trending__item__header"
        >
          @kendall
        </h2>
        <p
          className="trending__item__p"
        >
          Life Changing App
        </p>
      </div>
    </div>
    <div
      className="trending__item"
    >
      <img
        alt=""
        className="avatar--small"
        src="octopus_avatar.svg"
      />
      <div
        className="trending__item__text-div"
      >
        <h2
          className="trending__item__header"
        >
          @alex
        </h2>
        <p
          className="trending__item__p"
        >
          No Traffic Maker
        </p>
      </div>
    </div>
    <a
      className="trending__avatar-credit"
      href="https://www.vecteezy.com/free-vector/animal-avatar"
    >
      Animal Avatar Vectors by Vecteezy
    </a>
  </div>
</div>
`;
